src/concat.js

```diff
@@ -25,12 +25,12 @@ var _isArray = require('./internal/_isArray');
  *      R.concat([4, 5, 6], [1, 2, 3]); //=> [4, 5, 6, 1, 2, 3]
  *      R.concat('ABC', 'DEF'); // 'ABCDEF'
  */
-module.exports = _curry2(function(set1, set2) {
-  if (_isArray(set2)) {
-    return _concat(set1, set2);
-  } else if (_hasMethod('concat', set1)) {
-    return set1.concat(set2);
+module.exports = _curry2(function(a, b) {
+  if (_isArray(a) && _isArray(b)) {
+    return _concat(a, b);
+  } else if (_hasMethod('concat', a)) {
```

Comment:
I think we should check that both `a` and `b` have `concat` methods. If `a` has a `concat` method but `b` does not, they are not of the same type so we should throw a TypeError.